custom_components/car_wash/binary_sensor.py

```diff
@@ -157,7 +158,15 @@ async def async_update(self):
         tmpu = self.hass.config.units.temperature_unit
         temp = wdata.attributes.get(ATTR_WEATHER_TEMPERATURE)
         cond = wdata.state
-        forecast = wdata.attributes.get(ATTR_FORECAST)
+
+        daily_response = await self.hass.services.async_call(
+            "weather",
+            "get_forecasts",
```

Comment:
You might want to use `SERVICE_GET_FORECASTS` from https://github.com/home-assistant/core/blob/dev/homeassistant/components/weather/__init__.py#L137